packages/client/src/components/ProductCard/ProductCard.component.js

```diff
@@ -7,21 +7,33 @@ import StarRateIcon from '../../assets/star-rate-icon.svg';
 import BackgroundImage from '../../assets/jeans&shoes.jpg';
 import PropTypes from 'prop-types';
 
-const ProductCard = ({ id, title, price, isFavorite, toggleFavorite }) => {
+const ProductCard = ({
+  id,
+  title,
+  price,
+  isFavorite,
+  toggleFavorite,
+  setOpenModal,
+}) => {
   return (
     <div className="product-card-wrapper">
       <div className="product-img-wrapper">
         <img src={BackgroundImage} alt="Cart Icon" className="product-img" />
         <div className="cart-favorite-wrapper">
           <img src={CartIcon} alt="Cart Icon" />
           <button
-            onClick={(event) => toggleFavorite(id, title, price, event)}
+            onClick={(event) => {
+              isFavorite
+                ? setOpenModal({ modalStatus: true, favoriteId: id })
+                : toggleFavorite(id, title, price, event);
+            }}
             type="button"
             className="favorite-button"
           >
             <img
               src={isFavorite ? FullFavoriteIcon : EmptyFavoriteIcon}
               alt="Favorite Icon"
+              role="presentation"
```

Comment:
Not sure why you're adding this? I would avoid changing ARIA roles unless you really know what you are doing.

Reply:
That is the element for screen readers to understand but it should not be here so I will remove it, Thank you 😊